JavadocParagraph: define in documentation list of validated block-tags

https://checkstyle.org/checks/javadoc/javadocparagraph.html#Description

Warnings are generated about the presence (or lack of presence) of \<p> tag in javadoc text that is inconsistent with google-java-format-1.24.0.

```
$ cat Test.java 
package org.natuna.test;

/**
 * Plugin to standardize all directions to their appropriate abbreviation.
 *
 * <div style="padding: 1em 1em 1em">
 * <table>
 * <caption></caption>
 * <thead>
 * <tr>
 * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Direction
 * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Abbreviation
 * </tr>
 * </thead>
 * <tbody>
 * <tr>
 * <td>East</td> <td>E</td>
 * </tr>
 * </tbody>
 * </table>
 * </div>
 *
 * <p>This plugin assumes numbered streets have the correct suffix, when applicable.
 */
public class Test {
  /** Default constructor. */
  public Test() {
    super();
  }
}

$ java -jar checkstyle-10.20.0-all.jar -c /google_checks.xml Test.java 
Starting audit...
Audit done.

$ java -jar google-java-format-1.24.0-all-deps.jar Test.java > TestUpdated.java
$ cat TestUpdated.java 
package org.natuna.test;

/**
 * Plugin to standardize all directions to their appropriate abbreviation.
 *
 * <p><div style="padding: 1em 1em 1em">
 *
 * <table>
 * <caption></caption>
 * <thead>
 * <tr>
 * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Direction
 * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Abbreviation
 * </tr>
 * </thead>
 * <tbody>
 * <tr>
 * <td>East</td> <td>E</td>
 * </tr>
 * </tbody>
 * </table>
 *
 * </div>
 *
 * <p>This plugin assumes numbered streets have the correct suffix, when applicable.
 */
public class Test {
  /** Default constructor. */
  public Test() {
    super();
  }
}

$ java -jar checkstyle-10.20.0-all.jar -c /google_checks.xml TestUpdated.java 
Starting audit...
[WARN] /var/tmp/TestUpdated.java:6:4: <p> tag should not precede HTML block-tag
      '<div>', <p> tag should be removed. [JavadocParagraph]
[WARN] /var/tmp/TestUpdated.java:27:1: The name of the outer type and the file do not match. [OuterTypeFilename]
Audit done.
```

Expected: no violation `tag should not precede HTML block-tag '<div>', <p> tag should be removed. [JavadocParagraph]`
Violation from OuterTypeFilename is side effect for formatter output to new file.

```
$ cat Test.java 
public class Test {
    /**
     * Plugin to standardize all directions
     *
     * <p><dt>some text</dt> some text out
     *
     * <p>This plugin assumes numbered streets
     */
    public void plugin() {}
}

$ javadoc Test.java
Loading source file Test.java...
Constructing Javadoc information...
Building index for all the packages and classes...
Standard Doclet version 21.0.3+9-LTS
Building tree for all the packages and classes...
Generating ./Test.html...
Test.java:5: warning: empty <p> tag
     * <p><dt>some text</dt> some text out
       ^
Test.java:5: error: tag not allowed here: <dt>
     * <p><dt>some text</dt> some text out
          ^
Test.java:1: warning: no comment
public class Test {
       ^
Test.java:1: warning: use of default constructor, which does not provide a comment
public class Test {
       ^
Generating ./package-summary.html...
Generating ./package-tree.html...
Generating ./overview-tree.html...
Building index for all classes...
Generating ./allclasses-index.html...
Generating ./allpackages-index.html...
Generating ./index-all.html...
Generating ./search.html...
Generating ./index.html...
Generating ./help-doc.html...
1 error
3 warnings
```

so not all HTML bock-tags are supported by javadoc tool, we have to define exact list of tags.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java
@@ -41,9 +41,15 @@ import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
  * <li>There is one blank line between each of two paragraphs.</li>
  * <li>Each paragraph but the first has &lt;p&gt; immediately
  * before the first word, with no space after.</li>
- * <li>First paragraph tag should not precede
- * <a href="https://www.w3schools.com/html/html_blocks.asp">HTML block-tag</a>,
- * nested paragraph tags are allowed to do that.</li>
+ * <li>The outer most paragraph tags should not precede
+ * <a href="https://www.w3schools.com/html/html_blocks.asp">HTML block-tag</a>.
+ * Nested paragraph tags are allowed to do that. This check only supports following block-tags:
+ * &lt;address&gt;,&lt;blockquote&gt;
+ * ,&lt;div&gt;,&lt;dl&gt;
+ * ,&lt;h1&gt;,&lt;h2&gt;,&lt;h3&gt;,&lt;h4&gt;,&lt;h5&gt;,&lt;h6&gt;,&lt;hr&gt;
+ * ,&lt;ol&gt;,&lt;p&gt;,&lt;pre&gt;
+ * ,&lt;table&gt;,&lt;ul&gt;.
+ * </li>
  * </ul>
  *
  * <p><b>ATTENTION:</b></p>

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/javadoc/JavadocParagraphCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/javadoc/JavadocParagraphCheck.xml
@@ -15,9 +15,15 @@
  &lt;li&gt;There is one blank line between each of two paragraphs.&lt;/li&gt;
  &lt;li&gt;Each paragraph but the first has &amp;lt;p&amp;gt; immediately
  before the first word, with no space after.&lt;/li&gt;
- &lt;li&gt;First paragraph tag should not precede
- &lt;a href="https://www.w3schools.com/html/html_blocks.asp"&gt;HTML block-tag&lt;/a&gt;,
- nested paragraph tags are allowed to do that.&lt;/li&gt;
+ &lt;li&gt;The outer most paragraph tags should not precede
+ &lt;a href="https://www.w3schools.com/html/html_blocks.asp"&gt;HTML block-tag&lt;/a&gt;.
+ Nested paragraph tags are allowed to do that. This check only supports following block-tags:
+ &amp;lt;address&amp;gt;,&amp;lt;blockquote&amp;gt;
+ ,&amp;lt;div&amp;gt;,&amp;lt;dl&amp;gt;
+ ,&amp;lt;h1&amp;gt;,&amp;lt;h2&amp;gt;,&amp;lt;h3&amp;gt;,&amp;lt;h4&amp;gt;,&amp;lt;h5&amp;gt;,&amp;lt;h6&amp;gt;,&amp;lt;hr&amp;gt;
+ ,&amp;lt;ol&amp;gt;,&amp;lt;p&amp;gt;,&amp;lt;pre&amp;gt;
+ ,&amp;lt;table&amp;gt;,&amp;lt;ul&amp;gt;.
+ &lt;/li&gt;
  &lt;/ul&gt;
 
  &lt;p&gt;&lt;b&gt;ATTENTION:&lt;/b&gt;&lt;/p&gt;

--- a/src/xdocs/checks/javadoc/javadocparagraph.xml
+++ b/src/xdocs/checks/javadoc/javadocparagraph.xml
@@ -24,9 +24,15 @@
             no space after.
           </li>
           <li>
-            First paragraph tag should not precede
-            <a href="https://www.w3schools.com/html/html_blocks.asp">HTML block-tag</a>,
-            nested paragraph tags are allowed to do that.
+            The outer most paragraph tags should not precede
+            <a href="https://www.w3schools.com/html/html_blocks.asp">HTML block-tag</a>.
+            Nested paragraph tags are allowed to do that.
+            This check only supports following block-tags:
+            &lt;address&gt;,&lt;blockquote&gt;
+            ,&lt;div&gt;,&lt;dl&gt;
+            ,&lt;h1&gt;,&lt;h2&gt;,&lt;h3&gt;,&lt;h4&gt;,&lt;h5&gt;,&lt;h6&gt;,&lt;hr&gt;
+            ,&lt;ol&gt;,&lt;p&gt;,&lt;pre&gt;
+            ,&lt;table&gt;,&lt;ul&gt;.
           </li>
         </ul>
         <p><b>ATTENTION:</b></p>

--- a/src/xdocs/checks/javadoc/javadocparagraph.xml.template
+++ b/src/xdocs/checks/javadoc/javadocparagraph.xml.template
@@ -24,9 +24,15 @@
             no space after.
           </li>
           <li>
-            First paragraph tag should not precede
-            <a href="https://www.w3schools.com/html/html_blocks.asp">HTML block-tag</a>,
-            nested paragraph tags are allowed to do that.
+            The outer most paragraph tags should not precede
+            <a href="https://www.w3schools.com/html/html_blocks.asp">HTML block-tag</a>.
+            Nested paragraph tags are allowed to do that.
+            This check only supports following block-tags:
+            &lt;address&gt;,&lt;blockquote&gt;
+            ,&lt;div&gt;,&lt;dl&gt;
+            ,&lt;h1&gt;,&lt;h2&gt;,&lt;h3&gt;,&lt;h4&gt;,&lt;h5&gt;,&lt;h6&gt;,&lt;hr&gt;
+            ,&lt;ol&gt;,&lt;p&gt;,&lt;pre&gt;
+            ,&lt;table&gt;,&lt;ul&gt;.
           </li>
         </ul>
         <p><b>ATTENTION:</b></p>

